core/ProfileDetailGenerator.py

```diff
@@ -8,20 +9,43 @@ def __init__(self, profiles, mapping_type_code, blacklistedValueSets, fields_to_
         self.profiles = profiles
         self.mapping_type_code = mapping_type_code
         self.fields_to_exclude = fields_to_exclude
-        self.simple_data_types = ["instant", "time", "date", "dateTime", "decimal", "boolean", "integer", "string", "uri", "base64Binary", "code", "id", "oid", "unsignedInt", "positiveInt", "markdown", "url", "canonical", "uuid"]
+        self.simple_data_types = ["instant", "time", "date", "dateTime", "decimal", "boolean", "integer", "string",
+                                  "uri", "base64Binary", "code", "id", "oid", "unsignedInt", "positiveInt", "markdown",
+                                  "url", "canonical", "uuid"]
+
+    def find_and_load_scruct_def_from_path(self, struct_def, path):
+
+        elements = struct_def['snapshot']["element"]
+
+        for elem in (elem for elem in elements if elem['id'] == path):
+
+            elem_type = elem['type']
 
-    def get_value_sets_for_code_filter(self, structDef):
+            for type_elem in (elem for elem in elem_type if re.search("Reference", elem['code'])):
+                struct_def_ref = type_elem['targetProfile'][0]
+
+                if struct_def_ref not in self.profiles:
+                    for profile in self.profiles:
+                        if struct_def_ref.split("/")[-1] == profile.split("/")[-1]:
```

Comment:
What if there are multiple profiles with the same URL suffix? Would this lead to unintended behavior?

Reply:
@paulolaup good question - the first would be picked- which is not exactly accurate
but this will resolve once we only resolve mii references and all profiles have at least one mii reference - see comment below

Reply:
There can be cases where the targetProfile points to different resource types. We don't have a substance profile atm, but if we did, there would be two legitimate MII-URLs in the array

```json
"targetProfile": [
  "http://hl7.org/fhir/StructureDefinition/Substance",
  "https://www.medizininformatik-initiative.de/fhir/core/modul-medikation/StructureDefinition/Medication"
]

Reply:
@julsas would the substance then also have a code analogous to the Medication?